src/custom/state/swap/extension.ts

```diff
@@ -31,6 +42,48 @@ function _constructTradePrice({ sellToken, buyToken, kind, price }: TradeExecuti
   return executionPrice
 }
 
+function _minimumAmountOutExtension(pct: Percent, trade: TradeWithFee) {
+  if (trade.tradeType === TradeType.EXACT_OUTPUT) {
+    return trade.outputAmount
+  }
+
+  const priceDisplayed = trade.executionPrice.invert().raw
+  const slippage = new Fraction('1').add(pct)
+  const slippagePrice = priceDisplayed.multiply(slippage)
+  const minPrice = new Price(
+    trade.executionPrice.quoteCurrency,
+    trade.executionPrice.baseCurrency,
+    slippagePrice.denominator,
+    slippagePrice.numerator
+  )
+
+  const minimumAmountOut = minPrice.invert().quote(trade.inputAmountWithFee)
+
+  return minimumAmountOut
+}
+
+function _maximumAmountInExtension(pct: Percent, trade: TradeWithFee) {
```

Comment:
extended versions of Uni's slippage calculations. The originals can be seen here:
https://github.com/Uniswap/uniswap-v2-sdk/blob/a0e880df2ebce40311740e3cc009ffb08438e704/src/entities/trade.ts#L189